engine/src/main/java/org/terasology/rendering/dag/RenderingGraph.java

```diff
@@ -0,0 +1,140 @@
+import org.terasology.rendering.dag.nodes.OutlineNode;
+import org.terasology.rendering.dag.nodes.OverlaysNode;
+import org.terasology.rendering.dag.nodes.PrePostCompositeNode;
+import org.terasology.rendering.dag.nodes.ShadowMapNode;
+import org.terasology.rendering.dag.nodes.SimpleBlendMaterialsNode;
+import org.terasology.rendering.dag.nodes.SkyBandsNode;
+import org.terasology.rendering.dag.nodes.ToneMappedSceneNode;
+import org.terasology.rendering.dag.nodes.WorldReflectionNode;
+
+import java.util.Collection;
+import java.util.Map;
+
+/**
+ * TODO: Add javadocs
+ */
+public class RenderingGraph {
+    private final Context context;
+    private Map<String, Node> nodeMap;
+    private StateManager stateManager;
+    private Collection<Node> topologicalOrder;
+    private Config config;
+    private boolean analyzed;
+
+    public RenderingGraph(Context context) {
+        this.context = context;
+        this.nodeMap = Maps.newLinkedHashMap(); // insert order is important for now
+        this.config = context.get(Config.class);
+        this.stateManager = StateManager.createDefault(context);
+        this.analyzed = false;
+    }
+
+    public static RenderingGraph createDefault(Context context) {
+        RenderingGraph pipeline = new RenderingGraph(context);
+        pipeline.add("shadowMap", ShadowMapNode.class);
+        pipeline.add("worldReflective", WorldReflectionNode.class);
+        pipeline.add("backdrop", BackdropNode.class);
+        pipeline.add("skyBands", SkyBandsNode.class);
+        pipeline.add("objectsOpaque", ObjectsOpaqueNode.class);
+        pipeline.add("chunksOpaque", ChunksOpaqueNode.class);
+        pipeline.add("chunksAlphaReject", ChunksAlphaRejectNode.class);
+        pipeline.add("overlays", OverlaysNode.class);
+        pipeline.add("firstPersonView", FirstPersonViewNode.class);
+        pipeline.add("lightGeometry", LightGeometryNode.class);
+        pipeline.add("directionalLights", DirectionalLightsNode.class);
+        pipeline.add("chunksRefractiveReflective", ChunksRefractiveReflectiveNode.class);
+        pipeline.add("outline", OutlineNode.class);
+        pipeline.add("ambientOcclusionPasses", AmbientOcclusionPassesNode.class);
+        pipeline.add("prePostComposite", PrePostCompositeNode.class);
+        pipeline.add("simpleBlendMaterials", SimpleBlendMaterialsNode.class);
+        pipeline.add("lightShafts", LightShaftsNode.class);
+        pipeline.add("initialPostprocessing", InitialPostProcessingNode.class);
+        pipeline.add("downSampleSceneAndUpdateExposure", DownSampleSceneAndUpdateExposureNode.class); // TODO: a rename needed :)
+        pipeline.add("toneMapping", ToneMappedSceneNode.class);
+        pipeline.add("bloomPasses", BloomPassesNode.class);
+        pipeline.add("blurPasses", BlurPassesNode.class);
+        pipeline.add("finalPostProcessing", FinalPostProcessingNode.class);
+        pipeline.analyze();
+        return pipeline;
+    }
+
+    public void analyze() {
```

Comment:
Conceptually, this should go in the RenderPipelineOptimizer, that's its role and the implementation will obviously different.